src/Microsoft.DotNet.Darc/src/Darc/Operations/GetBuildOperation.cs

```diff
@@ -33,22 +34,79 @@ public override async Task<int> ExecuteAsync()
             {
                 IRemote remote = RemoteFactory.GetBarOnlyRemote(_options, Logger);
 
-                Build build = await remote.GetBuildAsync(_options.Id);
-                if (build != null)
+                IEnumerable<Build> matchingBuilds = null;
+                if (_options.Id != 0)
                 {
-                    Console.Write(UxHelpers.GetTextBuildDescription(build));
+                    if (!string.IsNullOrEmpty(_options.BuildUri) ||
+                        !string.IsNullOrEmpty(_options.Repo) ||
+                        !string.IsNullOrEmpty(_options.Commit))
+                    {
+                        Console.WriteLine("--id should not be used with other options.");
+                        return Constants.ErrorCode;
+                    }
+
+                    matchingBuilds = new List<Build>() { await remote.GetBuildAsync(_options.Id) };
+                }
+                else if (!string.IsNullOrEmpty(_options.BuildUri))
+                {
+                    if (_options.Id != 0 || !string.IsNullOrEmpty(_options.Repo) || !string.IsNullOrEmpty(_options.Commit))
+                    {
+                        Console.WriteLine("--uri should not be used with other options.");
+                        return Constants.ErrorCode;
+                    }
+
+                    Console.WriteLine("This option is not yet supported.");
+                    return Constants.ErrorCode;
+                }
+                else if (!string.IsNullOrEmpty(_options.Repo) || !string.IsNullOrEmpty(_options.Commit))
+                {
+                    if (string.IsNullOrEmpty(_options.Repo) != string.IsNullOrEmpty(_options.Commit))
+                    {
+                        Console.WriteLine("--repo and --commit should be used together.");
+                        return Constants.ErrorCode;
+                    }
+                    else if (_options.Id != 0 || !string.IsNullOrEmpty(_options.BuildUri))
+                    {
+                        Console.WriteLine("--repo and --commit should not be used with other options.");
+                        return Constants.ErrorCode;
+                    }
+
+                    matchingBuilds = await remote.GetBuildsAsync(_options.Repo, _options.Commit);
                 }
                 else
                 {
-                    Console.WriteLine($"Could not find build with id '{_options.Id}'");
+                    Console.WriteLine("Please specify --id, --uri, or --repo and --commit to lookup a build.");
                     return Constants.ErrorCode;
                 }
 
+                // Print the build info.
+                if (!matchingBuilds.Any())
```

Comment:
This is technically a possible multiple enumeration of this enum (which can do weird things for some flavors of enumerations). They are almost certainly just Lists or Arrays under the hoods, so it probably doesn't matter.